### ISSUE TITLE
Enable bazel builds without --nocheck_visibility

### ISSUE THREAD

**Issue by hawkinsp** [openxla/xla#1323]
Currently bazel builds of XLA from the OpenXLA repository require disabling visibility checks via the `--nocheck_visibility` argument to Bazel. It would be great if we could fix this!

**Comment by GMNGeoffrey** [comment]
I think the fundamental issue here is that visibility is almost useless outside of a monorepo. We also have to build our integration with TF with `--nocheck_visibility` (related, https://github.com/bazelbuild/bazel/issues/16767)

**Comment by hawkinsp** [comment]
I think it's still useful to have visibility within each repository, e.g., for internal structuring purposes. The key is that we should open visibility to any public targets that cross repository boundaries.

**Comment by GMNGeoffrey** [comment]
Right, it's nice intra-project but totally breaks across projects. Which is fine in google3 because it's all one "project". I think the *concept* of visibility groups is useful, but the implementation is totally not set up for a moderately complex constellation of projects

**Comment by hawkinsp** [comment]
Any updates? It's unfortunate to break visibility for all the downstream users and their own internals.
